crates/sp-lightclient/src/lib.rs

```diff
@@ -137,16 +143,18 @@ pub struct HeaderImporter<Header, Store>(PhantomData<(Header, Store)>);
 
 impl<Header: HeaderT, Store: Storage<Header>> HeaderImporter<Header, Store> {
     /// Verifies header, computes consensus values for block progress and stores the HeaderExt.
-    pub fn import_header(
-        store: &mut Store,
-        mut header: Header,
-    ) -> Result<(), ImportError<HashOf<Header>>> {
+    pub fn import_header(store: &mut Store, mut header: Header) -> Result<(), ImportError<Header>> {
         // check if the header is already imported
         match store.header(header.hash()) {
             Some(_) => Err(ImportError::HeaderAlreadyImported),
             None => Ok(()),
         }?;
 
+        // only try and import headers above the finalized number
+        if header.number() <= store.finalized_header().header.number() {
+            return Err(ImportError::HeaderIsBelowArchivingDepth);
```

Comment:
Might as well add those two numbers to the error variant for easier debugging if it ever happens.